instrumentation/opentelemetry-instrumentation-asgi/src/opentelemetry/instrumentation/asgi/__init__.py

```diff
@@ -395,6 +444,17 @@ async def otel_send(message):
                         set_status_code(server_span, 200)
                         set_status_code(send_span, 200)
                     send_span.set_attribute("type", message["type"])
+                    if (
+                        server_span.kind == trace.SpanKind.SERVER
```

Comment:
We should check if server_span is a recording span before checking its kind. NonRecordingSpan do not contain kind attribute. (https://github.com/open-telemetry/opentelemetry-python-contrib/issues/998 was reported for wsgi based frameworks for the same)